server/cluster.go

```diff
@@ -796,3 +806,31 @@ func (c *Cluster) rehash(nodes []string) []string {
 
 	return ringKeys
 }
+
+// Iterates over sessions hosted on this node and for each session
+// sends "{pres term}" to all displayed topics.
+// Called immediately after Cluster.rehash().
+func (c *Cluster) invalidateRemoteSubs(ss *SessionStore) {
+	ss.lock.Lock()
+	defer ss.lock.Unlock()
+
+	for _, s := range ss.sessCache {
+		if s.proto == CLUSTER || len(s.remoteSubs) == 0 {
+			continue
+		}
+		s.remoteSubsLock.Lock()
+		var topicsToTerminate []string
+		var keysToDelete []string
+		for topic, remSub := range s.remoteSubs {
+			if remSub.node != c.ring.Get(topic) {
+				topicsToTerminate = append(topicsToTerminate, remSub.originalTopic)
+				keysToDelete = append(keysToDelete, topic)
+			}
+		}
+		for _, topic := range keysToDelete {
```

Comment:
I believe this is unnecessary. You can remove map entries inside the `range s.remoteSubs`. It's safe.

Reply:
There is no value in moving it into a separate loop. It's still under the same loop.

Reply:
Done.

Reply:
The bot complaining about `keysToDelete` is correct.

Reply:
Removed it.